qualtran/_infra/data_types.py

```diff
@@ -70,6 +71,14 @@ def get_classical_domain(self) -> Iterable[Any]:
         """Yields all possible classical (computational basis state) values representable
         by this type."""
 
+    @abc.abstractmethod
+    def to_bits(self, x) -> Iterable[int]:
+        """Yields individual bits corresponding to binary representation of x"""
+
+    @abc.abstractmethod
+    def from_bits(self, *bits: int):
```

Comment:
It seems a little more symmetric to me to have `bits: Iterable[int]` everywhere, to match the return type of to_bits.

Reply:
Having `bits: Iterable[int]` is annoying because then I won't be able to subscript into `bits`. I can change it to `bits: Sequence[int]` instead and return a `List[int]` instead of `Iterable[int]` from `to_bits` method so the return value is compatible with `from_bits`.

Reply:
done